### DISCUSSION TITLE
UniqueId

### DISCUSSION THREAD

**Discussion by JDCain** [xunit/xunit#2215] [Question] [answered]
In issue #1208 Brad says:
`The UniqueID property exists solely to compare test cases, especially as it applies to serialization. It came into being solely to support the architecture of VSTest, which does discovery in one process and execution in another.`

Does this mean that if I am writing a vstest logger or data collector the UniqueId property in events are originally generated by xUnit?

**Comment by bradwilson** [comment]
I honestly have no idea how our UniqueID correlates to the one provided by VSTest.

**Comment by JDCain** [comment, marked as answer]
@bradwilson I was able to track down that I was looking for here: https://github.com/xunit/xunit/blob/a8614d34999c889e1c8014f679de95d20eae1304/src/xunit.runner.visualstudio/Sinks/VsDiscoverySink.cs#L89

For anyone who finds this: xUnit overrides the default Id generation of TestPlatform by setting the Guid it's self based off the executor and  uniqueId. This explains to me why xUnit theories don't generate duplicate TestId's within a run by vstest.
